### PR TITLE
Virus change, AI behaviour levels and invasion hotfix

### PR DESCRIPTION
**Pull request by Booraz149**
**Small virus addition**
Virus, if grown at least twice it's size (hmax>=160) can split into smaller viruses.
The amound of viruses depends on the growth of the virus (although it is capped at 12 virus max)

**AI behaviour levels**
AI now has behaviour levels that are given to them on spawn
AI level now affects following:
Aim/ target snapping
Friendly fire return chance
Chance to fire

Levels 1-3 make the most difference, 3 being the strongest, while 0 is mostly default behaviour as it is now.

**Invasion hotfix**
Added a variable _invasion_spawns_con - Acts as a condition for the invasion
Basically, it's given a number (30 for now) that acts as a spawn counter and if it's above 0 the invasion cannot end.

**Notes:**
I couldn't make the virus that spawns when the large one is destroyed have the same color as the dead one.
I tried filter: this.filter in spawn parameters but it didn't seem to work.

### CODE REVIEW

**Comment by Booraz149** [comment]
Converted into draft for now, seems like there's a conflict somewhere.

**Comment by Booraz149** [comment]
Should be fixed now.

**Comment by Eric-Gurt** [comment]
I probably should have said that I already started to resolving conflicts some hours ago :P

**Comment by Booraz149** [comment]
Seems I unintentionally added and removed lines of code which caused the issues.
Guess I should know now if it happens again.
